js/misc/util.js

```diff
@@ -268,3 +269,67 @@ function blockClickEventsOnActor(actor) {
         return true;
     });
 }
+
+function getSearchEngineName() {
+    let path = GLib.build_filenamev([GLib.get_user_config_dir(), 'chromium', 'Default', 'Preferences']);
+    let parser = new Json.Parser();
+
+    try {
+        parser.load_from_file(path);
+    } catch(e if e.matches(GLib.FileError, GLib.FileError.NOENT)) {
+        // User has not run Chromium yet.
+        return null;
+    } catch (e) {
+        logError(e, 'error while parsing Chromium preferences');
+        return null;
+    }
+
+    let root = parser.get_root().get_object();
+
+    let searchProviderDataNode = root.get_member('default_search_provider_data');
+    if (!searchProviderDataNode || searchProviderDataNode.get_node_type() != Json.NodeType.OBJECT) {
+        /*
+         * Translators: this is the name of the search engine that shows in the
+         * Shell's desktop search entry.
+         */
+        return _("Google");
```

Comment:
Can't you make this a constant, so you don't have to hardcode Google all over the place?